packages/backend/src/plugins/catalog.ts

```diff
@@ -1,15 +1,58 @@
 // eslint-disable-next-line notice/notice
-import { CatalogBuilder } from '@backstage/plugin-catalog-backend';
+import {
+  CatalogBuilder,
+  EntityProvider,
+} from '@backstage/plugin-catalog-backend';
 import { ScaffolderEntitiesProcessor } from '@backstage/plugin-scaffolder-backend';
 import { Router } from 'express';
 import { PluginEnvironment } from '../types';
+import {
+  AWSLambdaFunctionProvider,
+  AWSS3BucketProvider,
+} from '@roadiehq/catalog-backend-module-aws';
+import { Duration } from 'luxon';
 
 export default async function createPlugin(
   env: PluginEnvironment,
 ): Promise<Router> {
+  type RunnableProvider = EntityProvider & {
+    run: () => Promise<void>;
+  };
   const builder = await CatalogBuilder.create(env);
+  const providers: RunnableProvider[] = [];
+  for (const config of env.config.getOptionalConfigArray(
```

Comment:
@kissmikijr is this what you were thinking on the config?

Reply:
tbh I didn't really think about, just saw that they use the config for providers. I think since this is fairly complex logic here we could put it in the readme

Reply:
done.